src/odemis/gui/cont/tabs.py

```diff
@@ -1352,19 +1352,24 @@ def _on_switch_btn(self, evt):
         # Note: s axis can only be moved when near engaged pos. So:
         #  * when parking/referencing => move s first, then l
         #  * when engaging => move l first, then s
+        # Some systems are special, default is overridden with MD_AXES_ORDER_REF
         mirror = self.tab_data_model.main.mirror
+        axes_order = mirror.getMetadata().get(model.MD_AXES_ORDER_REF, ("s", "l"))
+        if set(axes_order) != {"s", "l"}:
+            logging.warning("Axes order of mirror is %s, while should have s and l axes",
+                            axes_order)
         mstate = self._get_mirror_state(mirror)
+
+        moves = []
         if mstate == MIRROR_PARKED:
             # => Engage
-            f1 = (mirror.moveAbs, {"l": self._pos_engaged["l"]})
-            f2 = (mirror.moveAbs, {"s": self._pos_engaged["s"]})
-            moves = (f1, f2)
+            for a in reversed(axes_order):
```

Comment:
why reversed?

Reply:
If you read comment at l. 1352, it's explained.